drivers/dma/dma_stm32.c

```diff
@@ -198,20 +201,26 @@ static int dma_stm32_get_periph_increment(enum dma_addr_adj increment,
 static int dma_stm32_configure(struct device *dev, u32_t id,
 			       struct dma_config *config)
 {
+
 	struct dma_stm32_data *data = dev->driver_data;
-	struct dma_stm32_stream *stream = &data->streams[id];
 	const struct dma_stm32_config *dev_config =
 					dev->config->config_info;
 	DMA_TypeDef *dma = (DMA_TypeDef *)dev_config->base;
+	struct dma_stm32_stream *stream = &data->streams[id - STREAM_OFFSET];
 	LL_DMA_InitTypeDef DMA_InitStruct;
 	u32_t msize;
 	int ret;
 
+	/* give channel from index 0 */
+	id = id - STREAM_OFFSET;
+
 	if (id >= data->max_streams) {
+		LOG_ERR("cannot configure the dma stream %d.", id);
```

Comment:
Maybe here we use "channel index setup is not valid with value %d" is better?

Reply:
OK, The fact is that this "Id" refer to a dma stream. So I propose 
LOG_ERR("Channel %d is not valid for this dma.", id);